xarray/core/dataset.py

```diff
@@ -210,6 +224,7 @@ def __init__(self, data_vars=None, coords=None, attrs=None,
             coords = {}
         if data_vars is not None or coords is not None:
             self._set_init_vars_and_dims(data_vars, coords, compat)
+            self._check_multiindex_level_names()
```

Comment:
mmh not sure to understand as this checks for unique multi-index level names among all the (multi-index) coordinates of a new `DataArray` or a `Dataset` object. I guess we need to iterate over those coordinates... I can rename it if it's not clear.